File: ShoppingCart.Data/Services/UserSession/Basket.cs
using System.Collections.Generic;
using ShoppingCart.Core.Money;

namespace ShoppingCart.Data.Services.UserSession
{
    public class Basket
    {
        public Basket()
        {
            Items = new List<BasketItem>();
            Total = Money.From(0);
        }

        public List<BasketItem> Items { get; set; }
        public Money Total { get; set; }
        public bool AdjustedPrice { get; set; }
    }
}

File: ShoppingCart.Web/src/shared/services/user-basket/basket.ts
import {Money} from './money';
import {Pizza} from '../pizza/pizza';

export class Basket {
    constructor() {
        this.items = [];
        this.adjustedPrice = false;
    }

    adjustedPrice: boolean;
    total: Money;
    deal: {
        code: string;
    };
    items: Pizza[];
}
